src/CMakeLists.txt

```diff
@@ -312,6 +312,13 @@ else()
 	endif()
 endif()
 
+# On clang and gcc, some functionalities of std::atomic require -latomic.
+# See <https://en.cppreference.com/w/cpp/atomic/atomic#Notes>.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang"
+		OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+	set(PLATFORM_LIBS ${PLATFORM_LIBS} atomic)
+endif()
```

Comment:
This doesn't look safe without checking for library presence
or is this provided by the compiler in all cases?

Reply:
On my machine (arch, btw.), `libatomic.so` is contained in `gcc-libs`, the same package that contains gcc's `libstdc++.so`.
According to [here](https://clang.llvm.org/docs/Toolchain.html#atomics-library), there's also `compiler-rt` that provides a libatomic implementation. But the arch package just depends on `gcc-libs`.

So yes, it's probably always provided with the compilers' default runtimes, which should come with the compiler.

I guess it would be better to change the condition to link with `-latomic` if and only if `libstdc++` (gcc) or `libc++` (clang) is used. But I don't know how to check that.

FYI, I've also searched for a cmake module or so that does this easily, but only found some other projects that use their own thing, e.g. <https://sources.debian.org/src/mmseqs2/14-7e284%2Bds-1/cmake/FindAtomic.cmake/>, <https://gitlab.cern.ch/dss/eos/-/blob/master/cmake/FindAtomic.cmake>.